pkg/sentry/devices/nvproxy/version.go

```diff
@@ -119,11 +119,13 @@ type abiConAndChecksum struct {
 //     set are not versioned.
 //  4. allocation classes within NV_ESC_RM_ALLOC in frontend device (based on
 //     NVOS64_PARAMETERS.HClass).
+//
+// +stateify savable
 type driverABI struct {
-	frontendIoctl   map[uint32]frontendIoctlHandler
-	uvmIoctl        map[uint32]uvmIoctlHandler
-	controlCmd      map[uint32]controlCmdHandler
-	allocationClass map[uint32]allocationClassHandler
+	frontendIoctl   map[uint32]frontendIoctlHandler   `state:"nosave"`
+	uvmIoctl        map[uint32]uvmIoctlHandler        `state:"nosave"`
+	controlCmd      map[uint32]controlCmdHandler      `state:"nosave"`
+	allocationClass map[uint32]allocationClassHandler `state:"nosave"`
```

Comment:
Hmm drop the `nosave` annotations? Otherwise these maps will be `nil`. Are these required? Function pointers should be savable.

Reply:
Checkpointing panics if I don't add `nosave` to the functions (added error logs in attachment). All errors are similar:

```
unknown object (nvproxy.frontendIoctlHandler)(0xc28380): ...
unknown object (nvproxy.controlCmdHandler)(0xc2a420): ...
unknown object (nvproxy.allocationClassHandler)(0xc36780): ...
unknown object (nvproxy.uvmIoctlHandler)(0xc3cdc0): ...
```
[gpu_checkpointing.log](https://github.com/google/gvisor/files/13284109/gpu_checkpointing.log)

Reply:
Hmm such a restore would be useless (as the ABI is essentially empty).